kubeproxy/api/v1alpha1/kubeproxy_types.go

```diff
@@ -0,0 +1,67 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	addonv1alpha1 "sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/addon/pkg/apis/v1alpha1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// KubeProxySpec defines the desired state of KubeProxy
+type KubeProxySpec struct {
+	addonv1alpha1.CommonSpec `json:",inline"`
+	addonv1alpha1.PatchSpec  `json:",inline"`
+
+	ClusterCIDR string `json:"clusterCidr,omitempty"`
+}
+
+// KubeProxyStatus defines the observed state of KubeProxy
+type KubeProxyStatus struct {
+	addonv1alpha1.CommonStatus `json:",inline"`
+}
+
+// +kubebuilder:object:root=true
+
+// KubeProxy is the Schema for the  API
+type KubeProxy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   KubeProxySpec   `json:"spec,omitempty"`
+	Status KubeProxyStatus `json:"status,omitempty"`
+}
+
+var _ addonv1alpha1.CommonObject = &KubeProxy{}
```

Comment:
Sorry to comment to WIP PR.
Is it better to add `var _ addonv1alpha1.Patchable = &KubeProxy{}` ? 
Because of `KubeProxy` has `PatchSpec` interface.
As like [here](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/master/examples/dashboard-operator/api/v1alpha1/dashboard_types.go#L38).